test/iterable_basics_test.dart

```diff
@@ -407,6 +407,47 @@ void main() {
     });
   });
 
+  group('average', () {
+    test('returns average on list of numbers', () {
+      final nums = [2, 2, 4, 8];
```

Comment:
nit: this list is still inferred as a list of `<int>`s since there's no generic specified and the list only contains ints. 

Can you add generics so that this list is typed as a list of `<num>`s?

```dart
final nums = <num>[2, 2, 4, 8];
```

Reply:
Or, optionally, you could mix `int`s and `double`s in the list like this:

```dart
final nums = [1, 1.5, 2];
```

Reply:
Added int type generic and added new new test for mixed int and double.